src/core/connection_pool.c

```diff
@@ -0,0 +1,508 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Definitions for the MsQuic Connection Pool API, which allows clients to
+    create a pool of connections that are spread across RSS cores.
+
+--*/
+
+#include "precomp.h"
+
+#ifdef QUIC_CLOG
+#include "connection_pool.c.clog.h"
+#endif
+
+#define MAX_CONNECTION_POOL_RETRY_MULTIPLIER 2
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicConnPoolGetStartingLocalAddress(
```

Comment:
Please add a test ASAP so all the warnings about uncovered lines go away so I can review more easily 😄